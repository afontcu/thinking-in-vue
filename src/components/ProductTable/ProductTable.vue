<template>
  <table>
    <thead>
      <tr>
        <th align="left">Name</th>
        <th align="right">Price</th>
      </tr>
    </thead>
    <tbody>
      <ProductRow
        v-for="product in products"
        :key="product.id"
        :name="product.name"
        :price="product.price"
        :offer="product.offer"
      />
    </tbody>
  </table>
</template>

<script>
import ProductRow from '@/components/ProductRow'

export default {
  name: 'ProductTable',
  components: {
    ProductRow
  },
  props: {
    products: {
      type: Array,
      required: true
    }
  }
}
</script>

<style>
table, th, td {
  border-collapse: collapse;
  border: 0;
  padding: 1.2rem;
  border-top: 1px solid #e6e6e6;
}
table {
  width: 100%;
}
thead {
  background-color: #fafafa;
  font-size: 14px;
  font-weight: semibold;
  text-transform: uppercase;
  color: #4a4a4a;
}
</style>
