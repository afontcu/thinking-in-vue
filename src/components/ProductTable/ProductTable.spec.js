import ProductTable from './index.js'
import { mount } from '@vue/test-utils'

const propsData = {
  products: [
    {
      'id': 4450,
      'name': 'Producte 01',
      'offer': true,
      'price': 14.61
    },
    {
      'id': 2212,
      'name': 'Producte 02',
      'offer': true,
      'price': 21.02
    }]
}

describe('ProductTable.vue', () => {
  it('is a component', () => {
    const wrapper = mount(ProductTable, { propsData })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders an snapshot', () => {
    const wrapper = mount(ProductTable, { propsData })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
