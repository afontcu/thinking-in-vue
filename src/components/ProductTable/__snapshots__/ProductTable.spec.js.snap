// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable.vue renders an snapshot: default 1`] = `
<table>
    <thead>
        <tr>
            <th>Name</th>
            <th>Price</th>
        </tr>
    </thead>
    <tbody>
        <tr>
(??)
        </tr>
    </tbody>
</table>
`;
